### PR TITLE
BAAP-135: Basename IPFS avatar support

### PR DESCRIPTION
**Pull request by kirkas**
image

image

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/components/Basenames/UsernameAvatarField/index.tsx [hunk 1]
```diff
@@ -1,114 +1,171 @@
...
-
 export default function UsernameAvatarField({
+  onChangeFile,
   onChange,
   value,
   disabled = false,
   username,
 }: UsernameAvatarFieldProps) {
   const [error, setError] = useState<string>();
+  const [newValue, setNewValue] = useState<string>(value || '');
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/UsernameAvatarField/index.tsx:37]: prob shouldn't be named `value`

File: apps/web/src/components/Basenames/UsernameProfileEditModal/index.tsx [hunk 2]
```diff
@@ -179,6 +192,7 @@ export default function UsernameProfileEditModal({
             })
 
             .catch((error) => {
+              console.log(error);
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/UsernameProfileEditModal/index.tsx:195]: do you want to keep this?

File: apps/web/src/components/DropdownItem/index.tsx [hunk 3]
```diff
@@ -29,6 +29,11 @@ export default function DropdownItem({ children, copyValue, onClick }: DropdownI
     setLastCopiedId(id);
   }, [id, setLastCopiedId]);
 
+  const onClickHandler = useCallback(() => {
+    setOpen(false);
+    if (onClick) onClick();
```
Comment (JFrankfurt) [inline apps/web/src/components/DropdownItem/index.tsx:34]:
```suggestion
    onClick?();
```

p sure you can do this instead

**Review by JFrankfurt — APPROVED** [review]

File: apps/web/src/components/Basenames/UsernameProfileEditModal/index.tsx (hunk 2 above)
Comment (kirkas) [reply apps/web/src/components/Basenames/UsernameProfileEditModal/index.tsx:195]: Ooops

**Pull request merged**